lib/os/printk.c

```diff
@@ -122,6 +122,7 @@ void _vprintk(out_func_t out, void *ctx, const char *fmt, va_list ap)
 			case '-':
 				padding = PAD_SPACE_AFTER;
 				goto still_might_format;
+				break;
```

Comment:
Someone is going to go and remove these statements in the future when enough time has passed that people don't remember why they were added there. Now they look as weird as two consecutive return or break statements. So I'd suggest putting at least a code comment here so people understand that no matter how weird it looks they should not be removed.

Reply:
I find it baffling that 'goto' isn't an acceptable substitute for 'break' in this context.

Reply:
Me too. In fact, isn't it likely that compilers will complain about unreachable code in this case?

Reply:
@jhedberg your're right, this happens. The trade-off here is that we'd probably need deviation for dead code for "defensive programming" cases. But I agree with you guys, it is causing more hurt than helping. I'll remove these cases.